src/TTrigger.cpp

```diff
@@ -194,28 +199,32 @@ bool TTrigger::setRegexCodeList(QStringList regexList, QList<int> propertyList)
         if (propertyList.at(i) == REGEX_PERL) {
             const char* error;
             const QByteArray& regexp = regexList.at(i).toUtf8();
+            const QString& regexp16 = regexList.at(i);
 
             int erroffset;
 
             // PCRE_UTF8 needed to run compile in UTF-8 mode
             // PCRE_UCP needed for \d, \w etc. to use Unicode properties:
-            QSharedPointer<pcre> re(pcre_compile(regexp.constData(), PCRE_UTF8 | PCRE_UCP, &error, &erroffset, nullptr), pcre_deleter);
+            QSharedPointer<pcre> re(pcre_compile(regexp.data(), PCRE_UTF8 | PCRE_UCP, &error, &erroffset, nullptr), pcre_deleter);
+            // PCRE_UTF16 needed to run compile in UTF-16 mode
+            QSharedPointer<pcre16> re16(pcre16_compile(regexp16.utf16(), PCRE_UTF16 | PCRE_UCP, &error, &erroffset, nullptr), pcre16_deleter);
```

Comment:
If there IS an error then this code will put the error message in the SAME variable `(const char *) error` as the UTF-8 case - presumable the text will be similar but will it be identical (if it contains numeric values then it will not). OTOH is it possible for only ONE of them to error out but not the other?

`errorOffset` though - **does** refer to the regular expression - and if that does contain non-ASCII characters (and I think it will in practice) then it will be different between the UTF-8/UTF-16 cases. This would, I believe, show up as wrong indexes in error messages in the editor when non-ASCII patterns are entered. E.g. for:
```
^A ⛾ of coffee)$
```

The error would produce a message about an unmatched closing parenthesis and it would possibly locate it (in `erroffset`) at position 15 for the UTF-8 case and 13 for UTF-16 - so you will need the latter for the system message display in the Editor - however the `setError(...)` system for storing an error messages in the Mudlet item (Alias/Trigger/Keybinding etc.) is currently limited as it would use the same text for reporting back errors in the Lua system (via `nil` + error message run-time error reporting).  Sadly, THAT purpose will need indexes into UTF-8 texts.  We do not currently report PCRE compilation errors back from, e.g. `permRegexTrigger(...)` but if we did (and we should) that would need those - I have just raised an issue about that: https://github.com/Mudlet/Mudlet/issues/3039 .

tl;dr; At present the error messages from the PCRE libraries reported to the user (in the editor) need to relate to the UTF-16 form - and this PR currently will do that - we will probably want the UTF-8 ones eventually as they will be needed for the Lua sub-sytem then but not for the Editor as they were being used *before* this PR.

Reply:
I can see the problem, I haven't considered that.